pycbc/waveform/bank.py

```diff
@@ -394,24 +403,157 @@ def approximant(self, index):
+            self._tsc_short_template_mem = zeros(wav_len,
+                                                 dtype=numpy.complex64)
+
+        if match_test:
+            # Don't want to use INTERP waveforms in here
+            approximant = self.approximant(t_num)
+            if approximant.endswith('_INTERP'):
+                approximant = approximant.replace('_INTERP', '')
+            # Using SPAtmplt here is bad as the stored cbrt and logv get
+            # recalculated as we change delta_f values. Fall back to TaylorF2
+            # in lalsimulation.
+            if approximant == 'SPAtmplt':
+                approximant = 'TaylorF2'
+
+            if not hasattr(injections, 'short_truncated_injs'):
+                err_msg = 'To use a match threshold using coarse ' + \
+                          'waveforms to decide whether to compute ' + \
+                          'full matched filter, one must have stored ' + \
+                          'the injection waveforms.'
+                raise ValueError(err_msg)
+
+            try:
+                red_psd = self._tsc_short_psd_storage[id(segment.psd)]
+            except KeyError:
+                curr_psd = segment.psd.numpy()
+                step_size = int(trunc_delta_f/segment.psd.delta_f)
+                max_idx = int(trunc_f_max/segment.psd.delta_f)+1
+                red_psd_data = curr_psd[:max_idx:step_size]
+                red_psd = FrequencySeries(red_psd_data, delta_f=trunc_delta_f,
+                                          copy=False)
+                self._tsc_short_psd_storage[id(curr_psd)] = red_psd
+
+            # Generate short template
+            # WARNING: This is f-domain only! T-domain *may* work, but many
+            #          waveforms will refuse to generate.
+            if not t_num == self._tsc_short_template_id:
+                htilde = pycbc.waveform.get_waveform_filter(
+                    self._tsc_short_template_mem, self.table[t_num],
+                    approximant=approximant, f_lower=f_ref,
+                    f_final=trunc_f_max,
+                    delta_f=trunc_delta_f, distance=1./DYN_RANGE_FAC,
+                    delta_t=1./(2 * trunc_f_max))
+                self._tsc_short_template_id = t_num
+                self._tsc_short_template_wav = htilde
+            else:
+                htilde = self._tsc_short_template_wav
+
+        # Get times covered by segment analyze
+        seg_start_time = \
+            segment.cumulative_index / float(gwstrain.sample_rate) + \
+            start_time
+        seg_end_time = \
+            (segment.cumulative_index +
+             (segment.analyze.stop - segment.analyze.start)) / \
+            float(gwstrain.sample_rate) + start_time
+        # And add buffer
+        seg_start_time = seg_start_time - seg_buffer
+        seg_end_time = seg_end_time + seg_buffer
+
+        # Then loop over injections
+        filter_segment = False
```

Comment:
Using 'break' here reads strangely to me. Why not simply do the following?

``` 
# listed in order of precedence
if test1_on and test1_fails:
    return True
if test2_on and test2_fails:
   return True
return False
```

The case of *no* tests being activated should be caught at the beginning and should immediately return True.